src/KubernetesClient/Informers/KubernetesInformerOptions.cs

```diff
@@ -0,0 +1,17 @@
+namespace k8s.Informers
+{
+    public class KubernetesInformerOptions // theoretically this could be done with QObservable, but parsing expression trees is too much overhead at this point
+    {
+        /// <summary>
+        ///     The default options for kubernetes informer, without any server side filters
+        /// </summary>
+        public static KubernetesInformerOptions Default { get; } = new KubernetesInformerOptions();
```

Comment:
Given this type is not immutable, it's dangerous to expose a singleton; `Default.Namespace` could be accidentally mutated, thus changing the behaviour of the default shared informers.

Consider making it immutable and: 
1.  Add a ctor with optional arguments. There is precedence for this pattern, but it does not version well (i.e. adding a parameter is a breaking chance).
2. Add a fluent build. There is no precedence for this pattern, so it's worth gathering the thoughts of other's beforehand (@tg123 , @brendandburns).

Reply:
I've created a fluent builder, as I was planning to do it anyways for labels. Unit test coverage for builder is also added.